client/client_test.go

```diff
@@ -204,17 +224,111 @@ func (s *ClientSuite) TestNewTargets(c *C) {
 	c.Assert(files, HasLen, 0)
 }
 
+func (s *ClientSuite) TestLocalExpired(c *C) {
+	client := s.newClient(c)
+	duration := 100 * time.Millisecond
+	syncAndWait := func() {
+		s.syncLocal(c)
+		time.Sleep(duration)
```

Comment:
Instead of sleeping, I'd rather abstract the `time.Now()` calculation to be mutable, so for production do:

```go
var timeNow = func() time.Time { time.Now() }
```

And in tests you swap out `timeNow` for a `func()` that is ahead, which allows us to not sleep and make the tests that much faster.